core/shared/src/main/scala/fs2/StreamScope.scala

```diff
@@ -0,0 +1,49 @@
+package fs2
+
+
+/**
+  * Scope represents a controlled block of execution of the stream to track resources acquired and released during
+  * the interpretation of the Stream.
+  *
+  * Scope's methods are used to perform low-level actions on stream interpretation, such as leasing the resources.
+  */
+trait StreamScope[F[_]] {
```

Comment:
I'd prefer to call this `Scope` (and the one in `fs2.internal` could be called `ScopeImpl` or `DefaultScope` or something). Keep the uglier name for the internals and the nicer name for the API.

Reply:
Ok so how about Scope as public and ScopeRunFold in implementation?